airflow/models/dagrun.py

```diff
@@ -43,6 +44,20 @@
 from airflow.utils.types import DagRunType
 
 
+class _TISchedulingDecision(TypedDict):
```

Comment:
Is there a reason we need to use TypedDict? NamedTuple is much easier to use in many cases.

Reply:
No reason -- just happened to be the example we were looking at in PoolStats and copied that.